src/panel/direction/RouteResult.jsx

```diff
@@ -34,36 +35,32 @@ export default class RouteResult extends React.Component {
     });
   }
 
-  componentDidUpdate(prevProps) {
-    if (this.props.routes.length !== prevProps.routes.length) {
-      this.setState({ activeRouteId: 0 });
-    }
-  }
-
   selectRoute = routeId => {
-    if (routeId === this.state.activeRouteId) {
+    if (routeId === this.props.activeRouteId) {
       return;
     }
+
     Telemetry.add(Telemetry.ITINERARY_ROUTE_SELECT);
     fire('set_main_route', { routeId, fitView: true });
-    this.setState({ activeRouteId: routeId });
+
+    const search = updateQueryString({ selected: routeId });
+    window.app.navigateTo('routes/' + search, {}, { replace: true });
```

Comment:
This method doesn't get called when first rendering the results, so the url isn't updated unless you chose another alternative.
I wonder if this can be a problem. It may signal some inconsistency that could prove problematic later. What do you think?

Reply:
I'm in favor of trying to keeps things consistent. I updated the PR with an url update triggered in `didUpdate` lifecycle.